packages/host/app/lib/formatted-message/utils.ts

```diff
@@ -117,27 +119,40 @@ export interface HtmlTagGroup {
 }
 
 export function parseHtmlContent(htmlString: string): HtmlTagGroup[] {
-  const result: HtmlTagGroup[] = [];
-
-  // Regular expression to match <pre> tags and their content
-  const regex = /(<pre[\s\S]*?<\/pre>)|([\s\S]+?)(?=<pre|$)/g;
-
-  let match;
-  while ((match = regex.exec(htmlString)) !== null) {
-    if (match[1]) {
-      // This is a code block (pre tag)
-      result.push({
-        type: 'pre_tag',
-        content: match[1],
-      });
-    } else if (match[2] && match[2].trim() !== '') {
-      // This is non <pre> tag HTML
-      result.push({
-        type: 'non_pre_tag',
-        content: match[2],
-      });
+  let result: HtmlTagGroup[] = [];
+
+  // Using dom manipulation to find pre and non-pre tags
+  // because we need to avoid the nested html tags issue.
```

Comment:
```suggestion
  // We are creating a new element in the dom so that we can easily parse the content of the top level <pre> tags. Note that <pre> elements can have nested <pre> elements inside them and by querying the dom like that it's trivial to get its contents, compared to parsing the htmlString. 
```